TKGame/Components/Concrete/Player_PhysicsComponent.cs

```diff
@@ -10,25 +11,38 @@ namespace TKGame.Components.Concrete
 {
     internal class Player_PhysicsComponent : PhysicsComponent
     {
-        private static readonly float GRAVITY = 1.0f;
+        private const int GRAVITY = 1000;
+
         void PhysicsComponent.Update(Entity entity, GameTime gameTime/*, World &world*/) // The &reference isn't working.
         {
+            Player player = entity as Player;
             float deltaTime = (float)gameTime.ElapsedGameTime.TotalSeconds;
 
-            Vector2 endVel = entity.Velocity;
-            endVel += entity.MOVEMENT_SPEED * entity.Velocity * deltaTime;
-
-            endVel.Y += GRAVITY;
-
-            entity.Position += endVel;
-
-            entity.HitBox = new Rectangle(((int)entity.Position.X - ((int)entity.Size.X / 2)),
-                                            ((int)entity.Position.Y - (int)entity.Size.Y / 2),
-                                            (int)entity.Size.X,
-                                            (int)entity.Size.Y);
-
+            player.Position += player.Velocity * deltaTime;
+
+            if(player.CollidedVertically)
+            {
+                player.Velocity.Y = 0;
+                player.CollidedVertically = false;
+            } 
+            if(player.CollidedHorizontally)
+            {
+                // TODO
+            }
+            else
+            {
+                player.Velocity.Y += GRAVITY * deltaTime;
+            }
```

Comment:
Be careful here, when you set the Y vel to 0 and then assume that the entity won't some how move vertically again before the frame ends and this check doesn't happen again.